tests/translator/input/api_swagger_integration_with_string_api_id.yaml

```diff
@@ -0,0 +1,23 @@
+Resources:
+  HtmlFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: s3://sam-demo-bucket/member_portal.zip
+      Handler: index.gethtml
+      Runtime: nodejs12.x
+      Events:
+        GetHtml:
+          Type: Api
+          Properties:
+            RestApiId: HtmlApi
```

Comment:
Every usage I've seen and some documentation (e.g. https://github.com/aws/serverless-application-model/blob/develop/docs/cloudformation_compatibility.rst#api) seems to indicate that it only expects either a `Ref` intrinsic or a `string`. When I try running a local invoke with a `Sub`, I get: `samtranslator.model.exceptions.InvalidDocumentException: [InvalidResourceException('HtmlFunction', 'Event with id [GetHtml] is invalid. Api Event must reference an Api in the same template.')]`. 
I think this would be the case for all other intrinsics as well (https://github.com/aws/serverless-application-model/blob/5511198db567a3bae496c35fb44de8994597e5d7/samtranslator/plugins/api/implicit_rest_api_plugin.py#L88).

Reply:
That's fine but do we have specific tests for this? You are adding/touching a property and before we accept, we need to make sure we have tests that cover different intrinsic cases.